cluster/gce/gci/startup-script.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+  
+METADATA_ENDPOINT="http://metadata/computeMetadata/v1/instance/network-interfaces/0/ip-aliases/"
+METADATA_HEADER="Metadata-Flavor: Google"
+aliases=$(curl -s ${METADATA_ENDPOINT} -H "${METADATA_HEADER}")
+if [ ! -z $aliases ];
+then
+    ip=$(curl -s ${METADATA_ENDPOINT}0 -H "${METADATA_HEADER}")
```

Comment:
Actually, you should use the 1 ip-alias (which is the /32 internal ip address attached to master), but I don't like relying on the order that can change. An alternative would be to modify this line https://github.com/kubernetes/kubernetes/blob/50c082785657e9d45507635369b95862b3ebaafa/cluster/gce/util.sh#L2949
and, in addition to executing the script, also add it to /etc/init.d or to crontab to execute @ reboot. 

+ @mborsz who implemented this "attach internal ip" logic for his insights

Reply:
Are you sure it should be 1?
```
bartekzurawski1@kubernetes-master:~$ curl http://metadata/computeMetadata/v1/instance/network-interfaces/0/ip-alias
es/ -H "Metadata-Flavor: Google"
0

bartekzurawski1@kubernetes-master:~$ curl http://metadata/computeMetadata/v1/instance/network-interfaces/0/ip-alias
es/0 -H "Metadata-Flavor: Google"
10.128.0.19/32
```

About /etc/init.d and @reboot in crontab you mean to use it instead of startup-script metadata?

Reply:
Is there possibility that there will be more than one ip-alias assigned to master intsance?

Reply:
Yeah, if master is registered then you'll have at least 2 (one ip-range for pods running on master and the other for the internal master ip). See this master that was created with these scripts:
```
curl http://metadata/computeMetadata/v1/instance/network-interfaces/0/ip-aliases/ -H "Metadata-Flavor: Google"
0
1
mmatejczyk@e2e-big-master ~ $ curl http://metadata/computeMetadata/v1/instance/network-interfaces/0/ip-aliases/0 -H "Metadata-Flavor: Google"
10.64.0.0/24
mmatejczyk@e2e-big-master ~ $ curl http://metadata/computeMetadata/v1/instance/network-interfaces/0/ip-aliases/1 -H "Metadata-Flavor: Google"
10.40.0.2/32
```

That's why I don't think we should hardcode it. And because it's might be tricky to guess correctly which range is the "private ip address" (you could pick the one that has /32 mask, but it's also not bulletproof). 

We can either use crontab or /etc/init.d as I suggested before, or pass here the internal ip address directly. You'd have to check whether the internal ip is created at this point or not. If not then crontab or /etc/initi.d is most likely the way to go.

Reply:
I've changed it a little. Now script will check if it has set up metadata that contains ip-alias which is created during cluster spin up. It's also added into systemd as oneshot service.